Providers/Scripts/2.6x-2.7x/Scripts/nxOMSAgentNPMConfig.py

```diff
@@ -0,0 +1,215 @@
+    recvdContentChecksum = hashlib.md5(Contents).hexdigest()
+    if recvdContentChecksum != ContentChecksum:
+        # data is corrupt do not proceed further
+        LG().Log('ERROR', 'Content received did not match checksum, exiting Set')
+        return [-1]
+    (ConfigType, ConfigID, Contents, Ensure, ContentChecksum) = init_vars(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    retval = Set(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    return retval
+
+
+def Test_Marshall(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    recvdContentChecksum = hashlib.md5(Contents).hexdigest()
+    if recvdContentChecksum != ContentChecksum:
+        # data is corrupt do not proceed further
+        LG().Log('ERROR', 'Content received did not match checksum, exiting Test')
+        return [0]
+    (ConfigType, ConfigID, Contents, Ensure, ContentChecksum) = init_vars(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    retval = Test(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    return retval
+
+
+def Get_Marshall(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    arg_names = list(locals().keys())
+    (ConfigType, ConfigID, Contents, Ensure, ContentChecksum) = init_vars(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    retval = 0
+    retval = Get(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    ConfigType = protocol.MI_String(ConfigType)
+    ConfigID = protocol.MI_String(ConfigID)
+    Ensure = protocol.MI_String(Ensure)
+    Contents = protocol.MI_String(Contents)
+    ContentChecksum = protocol.MI_String(ContentChecksum)
+
+    retd = {}
+    ld = locals()
+    for k in arg_names:
+        retd[k] = ld[k]
+    return retval, retd
+
+
+############################################################
+# Begin user defined DSC functions
+############################################################
+
+def SetShowMof(a):
+    global show_mof
+    show_mof = a
+
+
+def ShowMof(op, ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    if not show_mof:
+        return
+    mof = ''
+    mof += op + ' nxOMSAgentNPMConfig MyNPMConfig \n'
+    mof += '{\n'
+    mof += '    ConfigType = "' + ConfigType + '"\n'
+    mof += '    ConfigID = "' + ConfigID + '"\n'
+    mof += '    Contents = "' + Contents + '"\n'
+    mof += '    Ensure = "' + Ensure + '"\n'
+    mof += '    ContentChecksum = "' + ContentChecksum + '"\n'
+    mof += '}\n'
+    f = open('./test_mofs.log', 'a')
+    Print(mof, file=f)
+    LG().Log('INFO', mof)
+    f.close()
+
+
+def Set(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    ShowMof('SET', ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    retval = 0
+    destFileFullPath = CONFIG_PATH.__add__(DEST_FILE_NAME)
+    if ConfigType != 'UpdateNPMAgentConfig':
+        LG().Log('ERROR', 'Config type did not match, exiting set')
+        return [retval]
+    if Ensure != 'Present':
```

Comment:
We don't need to take any action in case of 'Absent' here. Use of 'Absent' will be made to  remove our plugin and binaries when we add file update mechanism to the resource.